python/lsst/pipe/tasks/computeExposureSummaryStats.py

```diff
@@ -121,77 +121,24 @@ def run(self, exposure, sources, background):
-            # The calculation of effective psf area is taken from
-            # meas_base/src/PsfFlux.cc#L112. See
-            # https://github.com/lsst/meas_base/blob/
-            # 750bffe6620e565bda731add1509507f5c40c8bb/src/PsfFlux.cc#L112
-            psfArea = np.sum(im.array)/np.sum(im.array**2.)
-        else:
-            psfSigma = np.nan
-            psfIxx = np.nan
-            psfIyy = np.nan
-            psfIxy = np.nan
-            psfArea = np.nan
+        self.update_psf(summary, psf, bbox, sources)
 
         wcs = exposure.getWcs()
-        if wcs is not None:
-            sph_pts = wcs.pixelToSky(lsst.geom.Box2D(bbox).getCorners())
-            raCorners = [float(sph.getRa().asDegrees()) for sph in sph_pts]
-            decCorners = [float(sph.getDec().asDegrees()) for sph in sph_pts]
-
-            sph_pt = wcs.pixelToSky(bbox.getCenter())
-            ra = sph_pt.getRa().asDegrees()
-            decl = sph_pt.getDec().asDegrees()
-        else:
-            raCorners = [float(np.nan)]*4
-            decCorners = [float(np.nan)]*4
-            ra = np.nan
-            decl = np.nan
-
-        photoCalib = exposure.getPhotoCalib()
-        if photoCalib is not None:
-            zeroPoint = 2.5*np.log10(photoCalib.getInstFluxAtZeroMagnitude())
-        else:
-            zeroPoint = np.nan
-
         visitInfo = exposure.getInfo().getVisitInfo()
-        date = visitInfo.getDate()
+        self.update_wcs(summary, wcs, bbox, visitInfo)
 
-        if date.isValid():
-            # We compute the zenithDistance at the center of the detector rather
-            # than use the boresight value available via the visitInfo, because
-            # the zenithDistance may vary significantly over a large field of view.
-            observatory = visitInfo.getObservatory()
-            loc = EarthLocation(lat=observatory.getLatitude().asDegrees()*units.deg,
-                                lon=observatory.getLongitude().asDegrees()*units.deg,
-                                height=observatory.getElevation()*units.m)
-            obstime = Time(visitInfo.getDate().get(system=DateTime.MJD),
-                           location=loc, format='mjd')
-            coord = SkyCoord(ra*units.degree, decl*units.degree, obstime=obstime, location=loc)
-            with warnings.catch_warnings():
-                warnings.simplefilter('ignore')
-                altaz = coord.transform_to(AltAz)
-
-            zenithDistance = 90.0 - altaz.alt.degree
-        else:
-            zenithDistance = np.nan
+        photoCalib = exposure.getPhotoCalib()
+        self.update_photo_calib(summary, photoCalib)
 
         if background is not None:
             bgStats = (bg[0].getStatsImage().getImage().array
                        for bg in background)
-            skyBg = sum(np.median(bg[np.isfinite(bg)]) for bg in bgStats)
-        else:
-            skyBg = np.nan
+            summary.skyBg = sum(np.median(bg[np.isfinite(bg)]) for bg in bgStats)
```

Comment:
The initial default-construction of `summary = ExposureSummaryStats()` now initializes all of these to `None`.  In the other methods I've set things to `nan` explicitly because they might be called on objects that weren't just initialized that way (indeed, that's exactly what I'm planning to do on DM-35207), but here that's now unnecessary.